### PR TITLE
[BugFix] Apply length caps in check_stop even when min_tokens is not reached

### PR DESCRIPTION
**Pull request by BabyDrangoner**
## Purpose

Fixes a request hang in the V1 scheduler when `min_tokens` cannot be reached within the model's context window.

`check_stop()` in `vllm/v1/core/sched/utils.py` returns early while `num_output_tokens < min_tokens`. That early return was added in #26317 to keep EOS / stop tokens from ending a request too early, but it also skips the `max_model_len` / `max_tokens` length checks below it. So if a request hits `max_model_len` before reaching `min_tokens`, nothing ever stops it: the scheduler computes `num_new_tokens == 0` for it on every step and skips it, and the request sits in the running queue forever holding its KV blocks and a request slot. `llm.generate()` never returns and the engine core keeps spinning on an otherwise idle server.

This combination passes parameter validation, so it's easy to hit:

- explicitly, with `max_tokens == min_tokens` larger than the remaining context:

```python
from vllm import LLM, SamplingParams

llm = LLM(model="facebook/opt-125m", max_model_len=128, enforce_eager=True)
# hangs forever before this PR
llm.generate(["Hello, my name is"],
             SamplingParams(max_tokens=500, min_tokens=500, ignore_eos=True))
```

- or implicitly: `min_tokens <= max_tokens` is validated when `SamplingParams` is constructed, but with `max_tokens=None` the input processor later clamps `max_tokens` to `max_model_len - prompt_len`, which can end up below `min_tokens` with no re-check.

Per its docstring, `min_tokens` is only supposed to hold back EOS and `stop_token_ids`; the min-tokens logits processor and the stop-string check in the detokenizer already work that way. This PR moves the gate in `check_stop()` so it only defers the EOS / stop-token checks and the length caps always apply. When `min_tokens` is already satisfied the checks run in the same order as before, so behavior there is unchanged. Repetition detection stays behind the `min_tokens` gate as today.

I didn't find an existing issue or PR for this: #26317 introduced the early return but only considered EOS, #47489 is about terminal grammars under `min_tokens` and its current diff still skips the length caps below `min_tokens`, and #27640 was a pure reorder that was closed unmerged.

AI assistance (GitHub Copilot CLI) was used to prepare this PR.

## Test Plan

Added a regression test next to the existing `check_stop` min_tokens test. It fails on current main and passes with this fix:

```bash
pytest tests/v1/core/test_scheduler.py::test_check_stop_length_cap_below_min_tokens \
       tests/v1/core/test_scheduler.py::test_check_stop_min_tokens
```

Also ran on a single RTX PRO 6000 (torch 2.11.0+cu130):

```bash
pytest tests/v1/core/
pytest tests/v1/e2e/general/test_min_tokens.py
pytest tests/v1/engine/test_engine_core.py::test_engine_core_advanced_sampling \
       tests/v1/engine/test_engine_core.py::test_engine_core
pre-commit run --files vllm/v1/core/sched/utils.py tests/v1/core/test_scheduler.py
```

plus the repro script above and a quick decode-throughput A/B (opt-125m, 256 prompts x 200 tokens, 3 runs each) since `check_stop` runs per generated token.

## Test Result

- new regression test: fails on main, passes with the fix
- repro script: hangs indefinitely before (killed after 240s, engine spinning with 0 scheduled tokens); finishes with 122 tokens / `finish_reason=length` after
- `tests/v1/core/`: 463 passed, 1 failed — `test_async_scheduling_pp_allows_rescheduling_with_output_placeholders`, which fails identically on clean main
- `tests/v1/e2e/general/test_min_tokens.py`: 7 passed, 6 xpassed (stale xfail marks, see #47487)
- engine core tests: 2 passed
- pre-commit and `git diff --check`: clean
- throughput: 38.3k tok/s with the fix vs 38.9k without (median of 3), within run-to-run noise

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by BabyDrangoner** [comment]
@codex review

**Comment by BabyDrangoner** [comment]
cc @njhill — you reviewed the min_tokens stop-check change in #26317 and the finish_reason handling in #27555, so flagging this one: the min_tokens early return in `check_stop` also suppresses the max_model_len / max_tokens length caps, which lets a request with an unsatisfiable min_tokens spin in the running queue forever. Repro and before/after details in the description.
